opp_path_selection/src/path_selector.cpp

```diff
@@ -0,0 +1,299 @@
+    }    // end of for each neighbor
+  }      // end for each point in cloud
+  // graph G has now been created.
+
+  std::vector<int> point_seg_indices;
+  for (int i = 0; i < points.size(); i++)
+  {
+    double mind = 2000.0;
+    int minj = -1;
+    // TODO figure out why dmin is not equal to zero
+    for (int j = 0; j < input_cloud->points.size(); j++)
+    {
+      Eigen::Vector3d pt2(input_cloud->points[j].x, input_cloud->points[j].y, input_cloud->points[j].z);
+      Eigen::Vector3d D = pt2 - points[i];
+      if (D.norm() < mind)
+      {
+        mind = D.norm();
+        minj = j;
+      }
+    }
+    point_seg_indices.push_back(minj);
+  }
+
+  // find id of every point from marker array, needed for Dijkstras
+  // TODO this is the exact same code as above in other implementation, it should be a subroutine
+  std::vector<int> path_indices;
+  for (int i = 1; i < point_seg_indices.size(); i++)  // for each source path segment
+  {
+    // use dijkstras to find shortest connected path between every line segement defined by points
+    std::vector<double> d;
+    std::vector<vertex_des> p;
+    size_t num_verts = num_vertices(G);
+    size_t num_edges = boost::num_edges(G);
+    p.resize(num_verts);
+    d.resize(num_verts);
+    boost::property_map<MeshGraph, boost::edge_weight_t>::type weightmap = boost::get(boost::edge_weight, G);
+    boost::property_map<MeshGraph, boost::vertex_index_t>::type index_map = get(boost::vertex_index, G);
+    vertex_des start_vertex = vertex(point_seg_indices[i - 1], G);
+    boost::dijkstra_shortest_paths(
+        G,             // const Graph& g
+        start_vertex,  // vertex_descriptor s, source, all shortest path distances computed from here
+        &p[0],         // predecessorMap each vertex p[i] points the next closer one p[p[i]] the next after
+        &d[0],         // distanceMap result providing the distance to each vertex from s
+        weightmap,     // weightMap accesses the weight of each edge
+        index_map,     // takes a vertex descriptor and finds the index of the vertex
+        std::less<double>(),
+        boost::closed_plus<double>(),
+        1000.0,
+        //			      (std::numeric_limits<double>::max)(),
+        0.0,
+        boost::default_dijkstra_visitor());
+
+    int V = point_seg_indices[i];
+    int start_index = point_seg_indices[i - 1];
+    int q = 0;
+    // follow predicesor chain from end endex to start index, pushing each one onto the
+    std::vector<int> seg_indices;
+    while (V != start_index && q < p.size())
+    {
+      seg_indices.push_back(V);
+      V = p[V];
+      q++;
+    }
+    // TODO check that q != p.size(), in this case, there is no path between the two points in the graph
+    seg_indices.push_back(V);
+
+    // reverse order of each segment
+    for (int j = seg_indices.size() - 1; j >= 0; j--)
+    {
+      path_indices.push_back(seg_indices[j]);
+    }
+  }  // end of for each source path segment
+  return path_indices;
```

Comment:
Seems like you could probably share this code between both overloads of this function